mob.go

```diff
@@ -990,3 +1063,191 @@ func sayEmptyLine() {
+		err = nil
+	} else {
+		err = fmt.Errorf("%s were not listed in ~/.gitconfig. Try using fully qualified coauthors", strings.Join(missingAliases, ", "))
+	}
+
+	return coauthors, err
+}
+
+func loadCoauthorFromAlias(alias string) string {
+	return gitconfig(true, fmt.Sprintf("mob.%s", alias))
+}
+
+func writeCoauthorsToGitConfig(coauthors coauthors.CoauthorsMap) {
+	clearAndAnnounceClearStagedCoauthors()
+
+	if len(coauthors) == 0 {
+		return
+	}
+
+	allCoauthors := make([]string, 0, len(coauthors))
+	newCoauthorEmails := make([]string, 0, len(coauthors))
+	newCoauthorAliases := make([]string, 0, len(coauthors))
+
+	for alias, coauthor := range coauthors {
+		previous := loadCoauthorFromAlias(alias)
+		allCoauthors = append(allCoauthors, coauthor)
+		gitconfig(true, fmt.Sprintf("mob.staged.%s", alias), coauthor)
+
+		if previous != coauthor {
+			gitconfig(true, fmt.Sprintf("mob.%s", alias), coauthor)
+			if previous != "" {
+				sayInfo(fmt.Sprintf("mob alias `%s` was updated to refer to `%s`", alias, coauthor))
+			} else {
+				newCoauthorEmails = append(newCoauthorEmails, coauthor)
+				newCoauthorAliases = append(newCoauthorAliases, alias)
+			}
+		}
+	}
+
+	if len(newCoauthorAliases) > 0 {
+		var beingVerb string
+		if len(newCoauthorAliases) == 1 {
+			beingVerb = "was"
+		} else {
+			beingVerb = "were"
+		}
+
+		sayInfo(fmt.Sprintf("%s %s saved to ~/.gitconfig", strings.Join(newCoauthorEmails, ", "), beingVerb))
+		sayIndented(fmt.Sprintf("Next time you can use `mob start --with \"%s\"`", strings.Join(newCoauthorAliases, ", ")))
+	}
+
+	var presentPassiveVerb string
+	if len(allCoauthors) == 1 {
+		presentPassiveVerb = "has"
+	} else {
+		presentPassiveVerb = "have"
+	}
+
+	sayInfo(fmt.Sprintf("%s %s been staged as coauthors in ~/.gitconfig", strings.Join(allCoauthors, ", "), presentPassiveVerb))
+	sayIndented("They will appear as co-authors on your next WIP commit,")
+	sayIndented("and they will appear as co-authors after `mob done`.")
+}
+
+func appendCoauthorsToSquashMsg(workingDir string) error {
+	squashMsgPath := path.Join(workingDir, ".git", "SQUASH_MSG")
+	file, err := os.OpenFile(squashMsgPath, os.O_APPEND|os.O_RDWR, 0644)
+	if err != nil {
+		if err == os.ErrNotExist {
+			// No wip commits, nothing to squash, this isn't really an error
+			return nil
+		}
+		return err
+	}
+
+	defer file.Close()
+
+	topLevelAuthor := ""
+	coauthorsHashSet := make(map[string]bool)
+
+	authorOrCoauthorMatcher := regexp.MustCompile("(?i).*(author)+.+<+.*>+")
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if authorOrCoauthorMatcher.MatchString(line) {
+			author := stripToAuthor(line)
+
+			// committer of this commit should
+			// not be included as a co-author
+			if topLevelAuthor == "" || author == topLevelAuthor {
+				topLevelAuthor = author
+				continue
+			}
+			coauthorsHashSet[author] = true
+		}
+	}
+
+	if len(coauthorsHashSet) > 0 {
+		coauthors := make([]string, 0, len(coauthorsHashSet))
```

Comment:
would love a method that converts co authors to a git commit message string. So we can separate the logic from the file append.